tests/queries/0_stateless/00170_lower_upper_utf8.sql

```diff
@@ -27,3 +27,7 @@ select sum(lower(materialize('aaaaАБВГAAAAaaAA')) = materialize('aaaaАБВ
 select sum(upper(materialize('aaaaАБВГAAAAaaAA')) = materialize('AAAAАБВГAAAAAAAA')) = count() from system.one array join range(16384) as n;
 select sum(lowerUTF8(materialize('aaaaАБВГAAAAaaAA')) = materialize('aaaaабвгaaaaaaaa')) = count() from system.one array join range(16384) as n;
 select sum(upperUTF8(materialize('aaaaАБВГAAAAaaAA')) = materialize('AAAAАБВГAAAAAAAA')) = count() from system.one array join range(16384) as n;
+
+-- Turkish language
```

Comment:
I think you need to update the .reference file for this test, no?

Reply:
It would be interesting how the ICU-only processing affects performance. If ICU is really really slow, we might want to retain the current implementation under a setting (which is disabled by default).

So, perhaps you like to generate a test table with ASCII-only, with cyrillic-only, and with arbitrary-UTF8-strings and measure performance for the old and new implementations. Thanks!

Reply:
No significant performance degression was found 
```
select upperUTF8('ır ır ır ır ır ır ır ır ır ır ır ır ır ır ır ') from numbers(1000000000) format Null;

New:
0 rows in set. Elapsed: 0.405 sec. Processed 1.00 billion rows, 8.00 GB (2.47 billion rows/s., 19.73 GB/s.)
0 rows in set. Elapsed: 0.405 sec. Processed 1.00 billion rows, 8.00 GB (2.47 billion rows/s., 19.73 GB/s.)
0 rows in set. Elapsed: 0.410 sec. Processed 1.00 billion rows, 8.00 GB (2.44 billion rows/s., 19.49 GB/s.)

Old
0 rows in set. Elapsed: 0.383 sec. Processed 1.00 billion rows, 8.00 GB (2.61 billion rows/s., 20.87 GB/s.)
0 rows in set. Elapsed: 0.403 sec. Processed 1.00 billion rows, 8.00 GB (2.48 billion rows/s., 19.83 GB/s.)
0 rows in set. Elapsed: 0.386 sec. Processed 1.00 billion rows, 8.00 GB (2.59 billion rows/s., 20.73 GB/s.)
```

Reply:
``` sql 
select upperUTF8('skshpfdhpwoehowehoehfopehopehop') from numbers(1000000000) format Null;
New:
0 rows in set. Elapsed: 0.408 sec. Processed 1.00 billion rows, 8.00 GB (2.45 billion rows/s., 19.63 GB/s.)
0 rows in set. Elapsed: 0.383 sec. Processed 1.00 billion rows, 8.00 GB (2.61 billion rows/s., 20.87 GB/s.)
0 rows in set. Elapsed: 0.387 sec. Processed 1.00 billion rows, 8.00 GB (2.58 billion rows/s., 20.65 GB/s.)
Old: 
0 rows in set. Elapsed: 0.384 sec. Processed 1.00 billion rows, 8.00 GB (2.60 billion rows/s., 20.83 GB/s.)
0 rows in set. Elapsed: 0.390 sec. Processed 1.00 billion rows, 8.00 GB (2.57 billion rows/s., 20.53 GB/s.)
0 rows in set. Elapsed: 0.389 sec. Processed 1.00 billion rows, 8.00 GB (2.57 billion rows/s., 20.56 GB/s.)
```